crates/sui-framework/sources/collectible/kiosk.move

```diff
@@ -0,0 +1,474 @@
+    /// basic kiosk functions unless some logic for this is implemented
+    /// in a third party module.
+    public fun set_owner(
+        self: &mut Kiosk, cap: &KioskOwnerCap, ctx: &TxContext
+    ) {
+        assert!(object::id(self) == cap.for, ENotOwner);
+        self.owner = sender(ctx);
+    }
+
+    // === Publisher functions ===
+
+    /// Register a type in the Kiosk system and receive an `TransferPolicyCap`
+    /// which is required to confirm kiosk deals for the `T`. If there's no
+    /// `TransferPolicyCap` available for use, the type can not be traded in
+    /// kiosks.
+    public fun new_transfer_policy_cap<T: key + store>(
+        pub: &Publisher, ctx: &mut TxContext
+    ): TransferPolicyCap<T> {
+        assert!(package::from_package<T>(pub), 0);
+        let id = object::new(ctx);
+        event::emit(TransferPolicyCapIssued<T> { id: object::uid_to_inner(&id) });
+        TransferPolicyCap { id }
+    }
+
+    /// Special case for the `sui::collectible` module to be able to register
+    /// type without a `Publisher` object. Is not magical and a similar logic
+    /// can be implemented for the regular `register_type` call for wrapped types.
+    public(friend) fun new_transfer_policy_cap_protected<T: key + store>(
+        ctx: &mut TxContext
+    ): TransferPolicyCap<T> {
+        let id = object::new(ctx);
+        event::emit(TransferPolicyCapIssued<T> { id: object::uid_to_inner(&id) });
+        TransferPolicyCap { id }
+    }
+
+    /// Destroy a TransferPolicyCap.
+    /// Can be performed by any party as long as they own it.
+    public fun destroy_transfer_policy_cap<T: key + store>(
+        cap: TransferPolicyCap<T>
+    ) {
+        let TransferPolicyCap { id } = cap;
+        object::delete(id);
+    }
+
+    // === Place and take from the Kiosk ===
+
+    /// Place any object into a Kiosk.
+    /// Performs an authorization check to make sure only owner can do that.
+    public fun place<T: key + store>(
+        self: &mut Kiosk, cap: &KioskOwnerCap, item: T
+    ) {
+        assert!(object::id(self) == cap.for, ENotOwner);
+        self.item_count = self.item_count + 1;
+        dof::add(&mut self.id, Key { id: object::id(&item) }, item)
+    }
+
+    /// Take any object from the Kiosk.
+    /// Performs an authorization check to make sure only owner can do that.
+    public fun take<T: key + store>(
+        self: &mut Kiosk, cap: &KioskOwnerCap, id: ID
+    ): T {
+        assert!(object::id(self) == cap.for, ENotOwner);
+        self.item_count = self.item_count - 1;
+        df::remove_if_exists<Offer, u64>(&mut self.id, Offer { id });
+        dof::remove(&mut self.id, Key { id })
+    }
+
+    // === Trading functionality ===
+
+    /// Make an offer by setting a price for the item and making it publicly
+    /// purchasable by anyone on the network.
+    ///
+    /// Performs an authorization check to make sure only owner can sell.
+    public fun make_offer<T: key + store>(
```

Comment:
Offer wording is a bit confusing here -- I expect other people to make offers on an existing listing, and the owner to `make_listing` perhaps?

Also, this leads to a follow-up question -- what's the scenario in which someone puts an item in a kiosk, but doesn't make a listing out of it?

Reply:
It is possible that some folks would use Kiosks as the container for their objects - transfer the whole Kiosk or store your Collectibles in a marketplace's kiosk. So I wouldn't discard the case as we don't know what could be achieved here.

Reply:
Works for me!  Could you add these examples to the doc comment for `place` so other folks who have the same question will see them as well?